Lux/src/image_loader.cpp

```diff
@@ -0,0 +1,26 @@
+// image_loader wraps low level image I/O routines. Implementation can be changed without affecting image classes.
+// Using stb_image, C-style pointer is created when class is initialized with a file name, and freed in the destructor.
+// Copy constructor and operator= are disabled to prevent segmentation faults or memory leaks
+
+#define STB_IMAGE_IMPLEMENTATION
+#include "../../stb_image/stb_image.h"
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include "../../stb_image/stb_image_write.h"
+#include "image_loader.hpp"
+
+image_loader :: image_loader( const std::string& filename ) {
+    stbi_ptr = stbi_load( filename.c_str(), &xsiz, &ysiz, &channels, 0 );
+    if(!stbi_ptr) 
+        throw std::runtime_error (std::string( "Image load error: \nFile" ) + filename + "\nReason:" + stbi_failure_reason() + "\n" ); 
+    img = std::span< unsigned char >( stbi_ptr, xsiz * ysiz * 3 ); 
```

Comment:
can move this into colon area as well using ternary ? : operator

Reply:
I attempted this using the following code:
```image_loader :: image_loader( const std::string& filename ) : stbi_ptr( stbi_load( filename.c_str(), &xsiz, &ysiz, &channels, 0 ) ),
img( stbi_ptr ? std::span< unsigned char >( stbi_ptr, xsiz * ysiz * 3 ) : 
            throw std::runtime_error (std::string( "Image load error: \nFile" ) + filename + "\nReason:" + stbi_failure_reason() + "\n" ) ) {}```

I received the following compile time warnings:
```/Users/joyhughes/src/Jen/Lux/src/image_loader.cpp:12:55: warning: field 'xsiz' is uninitialized when used here [-Wuninitialized]
img( stbi_ptr ? std::span< unsigned char >( stbi_ptr, xsiz * ysiz * 3 ) : 
                                                      ^
/Users/joyhughes/src/Jen/Lux/src/image_loader.cpp:12:62: warning: field 'ysiz' is uninitialized when used here [-Wuninitialized]
img( stbi_ptr ? std::span< unsigned char >( stbi_ptr, xsiz * ysiz * 3 ) : 
                                                             ^
2 warnings generated.
```

Reply:
I received the following compile time warnings:
```/Users/joyhughes/src/Jen/Lux/src/image_loader.cpp:12:55: warning: field 'xsiz' is uninitialized when used here [-Wuninitialized]
img( stbi_ptr ? std::span< unsigned char >( stbi_ptr, xsiz * ysiz * 3 ) : 
                                                      ^
/Users/joyhughes/src/Jen/Lux/src/image_loader.cpp:12:62: warning: field 'ysiz' is uninitialized when used here [-Wuninitialized]
img( stbi_ptr ? std::span< unsigned char >( stbi_ptr, xsiz * ysiz * 3 ) : 
                                                             ^
2 warnings generated.
```